### DISCUSSION TITLE
客户端授权使用内置数据库时,设置的规则是否有顺序之分?

### DISCUSSION THREAD

**Discussion by trdmm** [emqx/emqx#14411] [Q&A] [answered]
比如配置**所有用户**的topic发布权限有如下几条:
- `/dtu/${username}/data` (Allow)
- `/dtu/${username}/status` (Allow)
- `#` (Deny)
- `/dtu/${username)/info` (Allow)

此时如果发布 `/dtu/${username)/info` 这个topic的消息时, 能否成功发送?

**Comment by id** [comment, marked as answer]
After receiving the client's operation request, EMQX matches the authorization rules in order from top to bottom. After successfully matching a rule, EMQX allows or denies the current request according to the setting, and stops matching subsequent rules.
